lib/rules/require-valid-css-selector-in-test-helpers.js

```diff
@@ -0,0 +1,158 @@
+  },
+  other: {
+    hasError: (selector) => !_isValidSelector(selector),
+    fix() {},
+    errorMessage: 'Syntax error, "{{selector}}" is not a valid selector',
+  },
+});
+
+function _isValidSelector(selector) {
+  try {
+    cssTree.parse(selector, {
+      context: 'selector',
+    });
+  } catch {
+    return false;
+  }
+
+  return true;
+}
+
+function _isAssertDomCall(node, assertIdentifierName) {
+  const calleeName = parseCallee(node) || [];
+
+  if (calleeName[1] !== 'dom' || !assertIdentifierName) {
+    return false;
+  }
+
+  return calleeName[0] === assertIdentifierName;
+}
+
+function _isQuerySelectorCall(node, isTestCall) {
+  const calleeName = parseCallee(node);
+
+  // only validate querySelector calls in the test context
+  if (!isTestCall) {
+    return false;
+  }
+
+  return QUERY_SELECTOR_METHODS.has(calleeName[1]) && PARENT_NODE_NAMES.has(calleeName[0]);
+}
+
+function _isTestHelperCall(node, hasTestHelperImport, localImportNames) {
+  const calleeName = parseCallee(node) || [];
+
+  return hasTestHelperImport && localImportNames.includes(calleeName[0]);
+}
+
+module.exports = {
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'disallow using invalid CSS selectors in test helpers',
+      category: 'Testing',
+      recommended: false,
+      url:
+        'https://github.com/ember-cli/eslint-plugin-ember/tree/master/docs/rules/require-valid-css-selector-in-test-helpers.md',
+    },
+    fixable: 'code',
+    schema: [],
+    messages: Object.keys(SELECTOR_RULES).reduce((accumulator, currentVal) => {
+      const _accumulator = accumulator || {};
+      _accumulator[currentVal] = SELECTOR_RULES[currentVal].errorMessage;
+
+      return _accumulator;
+    }, null),
+  },
+  create(context) {
+    let hasTestHelperImport = false;
+    let localImportNames = [];
+    let testImportLocalName = '';
+    let isTestCall = false;
+    let assertIdentifierName = '';
+
+    return {
+      'ImportDeclaration[source.value="@ember/test-helpers"]'(node) {
+        hasTestHelperImport = getPropertyValue(node, 'specifiers').find((specifier) =>
+          TEST_HELPER_IMPORTS.has(getPropertyValue(specifier, 'imported.name'))
+        );
+
+        localImportNames = getPropertyValue(node, 'specifiers')
+          .filter((specifier) =>
+            TEST_HELPER_IMPORTS.has(getPropertyValue(specifier, 'imported.name'))
+          )
+          .map((specifier) => getPropertyValue(specifier, 'local.name'));
+      },
+      'ImportDeclaration[source.value="qunit"]'(node) {
+        const testImportSpecifier = getPropertyValue(node, 'specifiers').find(
+          (specifier) => getPropertyValue(specifier, 'imported.name') === 'test'
+        );
+
+        if (testImportSpecifier) {
+          testImportLocalName = getPropertyValue(testImportSpecifier, 'local.name');
+        }
+      },
+      CallExpression(node) {
+        if (node.callee.name === testImportLocalName) {
+          isTestCall = true;
```

Comment:
Instead of storing a boolean for `isTestCall`, you should store the node in a variable called `currentTestNode`. Then in `CallExpression:exit` you can clear the node if the current node being exited matches.

Reply:
I've refactored this let me know what you think, thanks.